hy/core/macros.hy

```diff
@@ -20,6 +20,15 @@
          ~@(interleave (repeat name) rest))
      ~name))
 
+
+(defmacro assoc [coll k1 v1 &rest other-kvs]
+  (if (odd? (len other-kvs))
+    (macro-error (last other-kvs) "`assoc` takes an odd number of arguments"))
+  (setv c (if other-kvs (gensym "c") coll))
+  `(setv ~@(+ (if other-kvs [c coll] []) #*
+    (genexpr [`(get ~c ~k) v] [[k v] (partition (+ (, k1 v1) other-kvs))]))))
```

Comment:
`genexpr` is now indented at -7 from the opening bracket of its list. Line it up with the `(if`, or add a newline and line both the `(if` and `(genexpr` at +2.